src/org/omegat/core/segmentation/SRX.java

```diff
@@ -110,40 +105,105 @@ public SRX copy() {
+        saveToSrx (srx, outFile);
+    }
+    
+    public static void saveToSrx(SRX srx, File outFile) throws IOException {        
+        ObjectFactory factory = new ObjectFactory();
+        Srx jaxbObject = factory.createSrx();
+        jaxbObject.setVersion("2.0");
+        jaxbObject.setHeader (factory.createHeader());
+        jaxbObject.getHeader().setSegmentsubflows (srx.segmentSubflows ? "yes" : "no");
+        jaxbObject.getHeader().setCascade (srx.cascade ? "yes" : "no");  
+        jaxbObject.setBody (factory.createBody());
+        jaxbObject.getBody().setMaprules (factory.createMaprules());
+        jaxbObject.getBody().setLanguagerules (factory.createLanguagerules());
+        for (MapRule mr: srx.getMappingRules()) {
+            Languagemap map = new Languagemap(); 
+            map.setLanguagerulename(mr.getLanguage()); 
+            map.setLanguagepattern(mr.getPattern());
+            jaxbObject.getBody().getMaprules().getLanguagemap().add (map);
+            Languagerule lr = new Languagerule();
+            lr.setLanguagerulename(mr.getLanguage());
+            jaxbObject.getBody().getLanguagerules().getLanguagerule().add (lr);
+            for (Rule rule: mr.getRules()) {
+                gen.core.segmentation.Rule jaxbRule = factory.createRule();
+                lr.getRule().add(jaxbRule);
+                jaxbRule.setBreak (rule.isBreakRule() ? "yes" : "no");
+                jaxbRule.setBeforebreak(factory.createBeforebreak());
+                jaxbRule.getBeforebreak().setContent(rule.getBeforebreak());                
+                jaxbRule.setAfterbreak(factory.createAfterbreak());                
+                jaxbRule.getAfterbreak().setContent(rule.getAfterbreak());
+            }
+        }
+        
         try {
-            srx.setVersion(CURRENT_VERSION);
-            XMLEncoder xmlenc = new XMLEncoder(new FileOutputStream(outFile));
-            xmlenc.writeObject(srx);
-            xmlenc.close();
-        } catch (IOException ioe) {
+            Marshaller m = SRX_JAXB_CONTEXT.createMarshaller();
+            m.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, Boolean.TRUE);
+            m.marshal(jaxbObject, new FileOutputStream(outFile));
+        } catch (JAXBException ioe) {
             Log.logErrorRB("CORE_SRX_ERROR_SAVING_SEGMENTATION_CONFIG");
             Log.log(ioe);
-            throw ioe;
+            ioe.printStackTrace();
+            throw new IOException (ioe.getMessage());
         }
     }
-
+    
+    /**
+     * Loads the local segmentation file. Accepts SRX (default) or old CONF format.
+     * In case you use conf format, rules about old version remain valid.
+     **/
+    public static SRX loadSRX(File configDir) {
+        File inFile = null;
+        try { 
+            inFile = new File (configDir, SRX_SENTSEG); 
+            if (inFile.exists()) {
+                return loadSrxFile(inFile.toURI().toURL()); 
+            }
+        } catch (Exception o2) {
+            
```

Comment:
I agree, but before to do the change, small question: 
when SRX file is absent, we try to find CONF file instead, that's logical;
but if SRX is present but corrupted, should we fallback to CONF file, or to default? 
In case the file is present but corrupted, are you sure it is enough to put the info in the log? (or should we send a message through the GUI?)

Reply:
> In case the file is present but corrupted, are you sure it is enough to put the info in the log? (or should we send a message through the GUI?)

You're right, this is probably important enough to have a GUI message. But I definitely think we should not be reaching out to the GUI from this class: I have long wanted to modularize OmegaT, meaning splitting the GUI and internals into separate modules. The internals should not know about the GUI; this is definitely an internal class.

Instead the `loadSRX` method should throw an exception if the file is corrupted (or if it's missing, for that matter) and the caller should retry loading the defaults.

Reply:
> but if SRX is present but corrupted, should we fallback to CONF file, or to default?

To clarify on this point, I think we should not fall back to CONF; the road from CONF to SRX should be one-way only.

Reply:
> You're right, this is probably important enough to have a GUI message.
> Instead the loadSRX method should throw an exception

OK, I will study this possibility. This is important because now that we use a standard format, we are susceptible to load segmentation rules copied by the user from another CAT tool, and what is correctly read by one CAT tool may not work in another one...

I will write again once it is done.